### PR TITLE
feat: add lint-staged

### PR DESCRIPTION
**Pull request by Tanuj22**
Signed-off-by: Tanuj22 <tanujagarwal22@gmail.com>

**Description**

Add lint staged to run eslint only for files that are staged to be committed.
**Notes for Reviewers**

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: ui/package.json [hunk 1]
```diff
@@ -1,91 +1,95 @@
...
-    "eslint-plugin-jsx-a11y": "^6.3.1",
-    "eslint-plugin-react": "^7.20.5",
-    "eslint-plugin-react-hooks": "^4.0.8",
-    "http-proxy": "^1.18.1",
-    "husky": "^4.2.5",
-    "nodemon": "^2.0.4",
-    "start-server-and-test": "^1.11.2"
-  }
-}
+{
```
Comment (kushthedude) [inline ui/package.json:1]: Why there is a complete rewrite of the package.json? Can you please try to fix it and push only the relevant changes?

Comment (kushthedude) [reply ui/package.json:1]: Else it looks good!

Comment (Tanuj22) [reply ui/package.json:1]: This was my second attempt at it. Not sure why it's showing complete rewrite 🤔

Comment (kushthedude) [reply ui/package.json:1]: I guess this is the issue with your editor configuration, try once disabling your editor formatting else let's go ahead with it

Comment (Tanuj22) [reply ui/package.json:1]: FIxed in #1805

**Pull request closed without merging**